src/main/java/org/skriptlang/skript/bukkit/potion/LegacyPotionEffectTypeInfo.java

```diff
@@ -0,0 +1,107 @@
+package org.skriptlang.skript.bukkit.potion;
+
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.classes.Parser;
+import ch.njol.skript.classes.Serializer;
+import ch.njol.skript.lang.ParseContext;
+import ch.njol.skript.localization.Language;
+import ch.njol.util.StringUtils;
+import ch.njol.yggdrasil.Fields;
+import org.bukkit.potion.PotionEffectType;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.StreamCorruptedException;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+final class LegacyPotionEffectTypeInfo {
+
+	private static Map<String, PotionEffectType> types;
+	private static Map<String, String> names;
+
+	private static void init() {
+		types = new HashMap<>();
+		names = new HashMap<>();
+		Language.addListener(() -> {
+			types.clear();
+			names.clear();
+			for (PotionEffectType potionEffectType : PotionEffectType.values()) {
+				String key = potionEffectType.getKey().getKey();
+				String[] entries = Language.getList("potion effect types." + key);
+				names.put(key, entries[0]);
+				for (String entry : entries) {
+					types.put(entry.toLowerCase(Locale.ENGLISH), potionEffectType);
+				}
+			}
+		});
+	}
+
+	private LegacyPotionEffectTypeInfo() { }
+
+	public static ClassInfo<PotionEffectType> getInfo() {
+		init();
```

Comment:
Is this a necessary recomputation? Additionally are there possible adverse effects to this accidentally being called more than once?